packages/frontend/src/PortalForm/index.tsx

```diff
@@ -70,10 +70,18 @@ const getError = (name: string, errors: InputError[]): string | null => {
 
 const MappingBar = () => {
   const fromId = useSelector(
-    (state: RootState) => state.portalMap.inspectPortalId
+    (state: RootState) => state.portalMap.inspectFromId
   )
+  const toId = useSelector((state: RootState) => state.portalMap.inspectToId)
+
+  const size = useSelector((state: RootState) => state.portalMap.size)
+
+  const timeLeft = useSelector((state: RootState) => state.portalMap.timeLeft)
```

Comment:
So, I'm not sure what the problem with the map is, but you're doing both things here. 

You should rely on the selectors to update your data instead of the useEffects. I feel like you might be rendering more times than you intend to.

Reply:
The useSelector just gets the data from the RootStat. Then useEffect fires when this is changed and then actually changes what is displayed. The useSelecter doesn't change anything visually. And I also followed how it was done before for the from part of the form